spec.html

```diff
@@ -32544,67 +32544,55 @@ <h1>Array.prototype.some ( _callbackfn_ [ , _thisArg_ ] )</h1>
       <emu-clause id="sec-array.prototype.sort">
         <h1>Array.prototype.sort ( _comparefn_ )</h1>
         <p>The elements of this array are sorted. The sort must be stable (that is, elements that compare equal must remain in their original order). If _comparefn_ is not *undefined*, it should be a function that accepts two arguments _x_ and _y_ and returns a negative value if _x_ &lt; _y_, zero if _x_ = _y_, or a positive value if _x_ &gt; _y_.</p>
-        <p>Upon entry, the following steps are performed to initialize evaluation of the `sort` function:</p>
+        <p>The following steps are taken:</p>
         <emu-alg>
           1. If _comparefn_ is not *undefined* and IsCallable(_comparefn_) is *false*, throw a *TypeError* exception.
           1. Let _obj_ be ? ToObject(*this* value).
           1. Let _len_ be ? ToLength(? Get(_obj_, `"length"`)).
-        </emu-alg>
-        <p>Within this specification of the `sort` method, an object, _obj_, is said to be <em>sparse</em> if the following algorithm returns *true*:</p>
-        <emu-alg>
-          1. For each integer _i_ in the range 0 &le; _i_ &lt; _len_, do
-            1. Let _elem_ be _obj_.[[GetOwnProperty]](! ToString(_i_)).
-            1. If _elem_ is *undefined*, return *true*.
-          1. Return *false*.
+          1. Let _items_ be a new empty List.
+          1. Let _k_ be 0.
+          1. Let _itemCount_ be 0.
+          1. Let _undefinedCount_ be 0.
+          1. Repeat, while _k_ &lt; _len_
+            1. Let _Pk_ be ! ToString(_k_).
+            1. Let _kPresent_ be ? HasProperty(_obj_, _Pk_).
+            1. If _kPresent_ is *true*, then
+              1. Let _kValue_ be ? Get(_obj_, _Pk_).
+              1. If _kValue_ is *undefined*, then
+                1. Set _undefinedCount_ to _undefinedCount_ + 1.
+              1. Else,
+                1. Set _items_[_itemCount_] to _kValue_.
+                1. Set _itemCount_ to _itemCount_ + 1.
+            1. Set _k_ to _k_ + 1.
+          1. Sort _items_ using an implementation-dependant sequence of calls to SortCompare.
```

Comment:
```suggestion
          1. Sort _items_ using an implementation-dependent sequence of calls to SortCompare.
```

Reply:
FWIW I also prefer this spelling. We just stuck to "dependant" because that's what the current spec uses.

Reply:
I believe the a spelling only applies in old British English and only when used as a noun; since here it’s an adjective, the e spelling is universally correct.

I’ll audit the other places in the spec and update them in a separate PR.

Reply:
(i can't find any uses of `dependant` in the current spec to follow up with)